### DISCUSSION TITLE
How can we call markdoc object outside of _app.tsx in nextjs project?

### DISCUSSION THREAD

**Discussion by huskyjp** [markdoc/markdoc#312] [Q&A] [answered]
I am trying to implement `markdoc` with nextjs framework. 
As explained in its docs, if we want to have documentation in the root page, we can call and set configs in `_app.tsx` like [so][1]

```
// pages/_app.tsx

import Layout from '../components/Layout';

export default function App({ Component, pageProps }) {
  return (
    <Layout frontmatter={pageProps.markdoc.frontmatter}>
      <Component {...pageProps} />
    </Layout>
  );
}
```

And in this case, the structure is like below.
```
.
├── markdoc
│   ├── functions.js
│   ├── nodes
│   │   ├── ...
│   │   ├── link.markdoc.js
│   │   └── index.js
│   └── tags
│       ├── ...
│       └── index.js
├── pages
    ├── docs
│   ├── _app.tsx
│   └── index.md
└── next.config.js
```

But I want to have normal page in my `_app.tsx & index.tsx` (i.e. root page) then load and show markdoc contents under the `docs` path. (eg. localhost:3000/docs)

Hers is my case.
```
.
├── markdoc
│   ├── functions.js
│   ├── nodes
│   │   ├── ...
│   │   ├── link.markdoc.js
│   │   └── index.js
│   └── tags
│       ├── ...
│       └── index.js
├── pages
    ├── docs
        ├── index.md <- where my first docs content is stored
        ├── index.tsx <- where I want to call initial markdoc object to load frontmatter
│   ├── _app.tsx
│   └── index.tsx <- where my root page is stored with normal Next/React
└── next.config.js
```

In this case, how can I call `markdoc` object under the `docs` path as it is done in `_app.tsx`? Also, where should `index.md` be placed in this case? If I place the index.md in the root, there will be two entry root `index.tsx` and `index.md`, and I am confused the correct structure.

  [1]: https://markdoc.dev/docs/nextjs

**Comment by mfix-stripe** [comment, marked as answer]
@huskyjp you can just call it in `_app.tsx` if the route starts with `/docs`. We do a similar check here: https://github.com/markdoc/docs/blob/main/pages/_app.js#L83

You shouldn't need both a `docs/index.md` _and_ a `docs/index.tsx` — only `docs/index.md`
